.github/workflows/extract-exercises.yml

```diff
@@ -0,0 +1,36 @@
+name: Extract exercises
+on: workflow_dispatch
+jobs:
+  extract:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install caex2
+      - name: Fetch the url for the lesson content github repository from metadata
+        uses: jannekem/run-python-script-action@v1
+        with:
+          script: |
+            import csv
+            from urllib.request import urlretrieve
+
+            with open('_data/data.csv', newline='') as csvfile:
+                reader = csv.DictReader(csvfile)
+                for row in reader:
+                    curriculum = row['curriculum']
+            link = f"https://raw.githubusercontent.com/esciencecenter-digital-skills/workshop-metadata/main/{curriculum}/lesson-url.md"
```

Comment:
I considered putting this in a separate file and adding error handling, but I think that would actually be more difficult to maintain than this. It is quite simple and you have everything that happens in one place.

Reply:
When something goes wrong (such as a wrong url in the metadata), it is hopefully clear from the resulting error messages what went wrong.